yt/odin/Dockerfile

```diff
@@ -0,0 +1,135 @@
+ARG ROOT="/ytsaurus"
```

Comment:
Let's put it into `yt/docker/odin`?

Reply:
Can you elaborate? This path is virtual, it exists only inside the docker layer. It shouldn't concern you normally.

Reply:
I'm talking about the file in general

Reply:
Oh, you mean move `yt/odin/Dockerfile` to `yt/docker/odin/Dockerfile`.

Do you feel that it is absolutely necessary? I think keeping dockerfiles close to their code is a better practice, as odin is a pretty separate thing from the rest of the code, and ytsaurus acts here as monorepo. Also, there are some parts of YT whose dockerfiles do not belong to `yt/docker`, e.g. SPYT dockerfiles are stored separately. Therefore, I'd keep yt/docker as a place with c++ code docker images (which are tightly related to each other), and store docker images of separate components like Odin separately.

Also, in current placement you can describe both Odin and the arguments of its docker image in the same README.md file (as I suggest in one of the comments in this PR).